### PR TITLE
ENG-4284 logging, update to JDBC unit tests, JSON schema validation for AMP stored request(s)

### PR DESCRIPTION
**Pull request by maddapper**
Currently, JDBC tests are broken due to external dependency on a Postgres DB.  This change removes that dependency and uses an embedded database instead.  

Also, request and response is now being logged through a logging filter.  Logging settings is set through the yaml config.  It can be disabled or enabled and a limiter value can be set to throttle logging.   Limiter value can be set (0.0f - 1.0f).  A limiter value of 1.0f will completely filter out all logs, while a limiter value of 0.0f will allow all logs to be written. A limiter value of 0.5f will filter out 50% of the logs. The reason limiter value was included is so that we don't blow-up disk space usage in our node(s). Typically, for diagnostic purposes, only a small sample of logs need to be analyzed.  Phased rollout is suggested, first with logging disabled, then if there are no issues, with logging enabled with a limiter of 0.99.  Gradually increase limiter over time depending on how much logging is needed and as filesystem capacity dictates.  WARN logging level or higher needed to generate logs.  ERROR and FATAL logging level(s) will suppress logging.  Local yaml config used for testing (see below).  

Added JSON schema validation test(s) and support for AMP endpoint logging.

spring:
  profiles: local
logging:
  level:
    root: warn
logging.filter:
  request:
    enabled: true
    limiter-value: 0.00f
vertx:
  worker-pool-size: 5
  uploads-dir: file-uploads
  init-timeout-ms: 10000
  http-server-instances: 1
auction:
  ad-server-currency: USD
settings:
  filesystem:
    settings-filename: sample/sample-app-settings.yaml
    stored-requests-dir: tmp/stored_requests
    stored-imps-dir: tmp/stored_imps
    stored-responses-dir: tmp/stored_responses

Examples of logging:

2020-07-02 11:28:59.295 ERROR 22483 --- [ntloop-thread-4] o.p.s.handler.SessionLoggingHandler      : 127.0.0.1 - - [Thu, 02 Jul 2020 18:28:59 GMT] "GET /openrtb2/amp?tag_id=Freestar_Test_320x500 HTTP/1.1" 400 77 "-" "insomnia/2020.2.2" [?]  [?] Invalid request format: No stored request found for id: Freestar_Test_320x500

2020-07-02 11:29:05.134  WARN 22483 --- [ntloop-thread-4] o.p.s.handler.SessionLoggingHandler      : 127.0.0.1 - - [Thu, 02 Jul 2020 18:29:04 GMT] "GET /openrtb2/amp?tag_id=Freestar_Test_320x50 HTTP/1.1" 200 16 "-" "insomnia/2020.2.2" [?]  [?] {"targeting":{}}

2020-07-02 11:29:11.736  WARN 22483 --- [ntloop-thread-4] o.p.s.handler.SessionLoggingHandler      : 0:0:0:0:0:0:0:1 - - [Thu, 02 Jul 2020 18:29:11 GMT] "POST /openrtb2/auction HTTP/1.1" 200 957 "-" "insomnia/2020.2.2" [?] {"device":{"w":414,"h":896,"ifa":"A74A7CB3-5B92-4C38-84AC-46D9535DCDAC","devtime":1570489572,"osv":"12.2","lmt":0,"connectiontype":1,"os":"iOS","geo":{"lastfix":30002,"lat":37.785834,"lon":-122.406417,"accuracy":5},"pxratio":2,"make":"Apple","ua":"Mozilla/5.0 (iPhone; CPU iPhone OS 12_2 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Mobile/15E148","model":"x86_64"},"source":{"tid":"123"},"app":{"bundle":"456805313","ext":{"prebid":{"version":"0.6.2","source":"prebid-mobile"}},"publisher":{"id":"88"},"ver":"2.0.0"},"id":"C3F4EB34-C0FA-4B10-8D6F-6C558E9B25DF","imp":[{"id":"Freestar_Test_320x50","ext":{"rubicon":{"accountId":16924,"siteId":222354,"zoneId":1141820},"rhythmone":{"placementId":"79261","path":"mvo","zone":"1r"},"districtm":{"placementId":15832560},"appnexus":{"placementId":13452978}},"secure":1,"banner":{"format":[{"w":320,"h":50}]}}],"ext":{"prebid":{"cache":{"bids":{}},"aliases":{"appnexusAst":"appnexus","districtm":"appnexus"},"targeting":{"pricegranularity":"high","lengthmax":20}}},"regs":{"ext":{"gdpr":0}},"user":{"yob":1995,"gender":"F"}} [?] {"id":"C3F4EB34-C0FA-4B10-8D6F-6C558E9B25DF","seatbid":[],"cur":"USD","nbr":0,"ext":{"errors":{"districtm":[{"code":2,"message":"appnexus is not configured properly on this Prebid Server deploy. If you believe this should work, contact the company hosting the service and tell them to check their configuration."}],"rubicon":[{"code":3,"message":"Unexpected status code: 401. Run with request.test = 1 for more info"}],"rhythmone":[{"code":2,"message":"rhythmone is not configured properly on this Prebid Server deploy. If you believe this should work, contact the company hosting the service and tell them to check their configuration."}],"appnexus":[{"code":2,"message":"appnexus is not configured properly on this Prebid Server deploy. If you believe this should work, contact the company hosting the service and tell them to check their configuration."}]},"responsetimemillis":{"districtm":0,"rhythmone":0,"appnexus":0,"rubicon":214},"tmaxrequest":900}}

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -55,6 +55,7 @@
         <wiremock.version>2.11.0</wiremock.version>
         <restassured.version>3.0.6</restassured.version>
         <h2.version>1.4.196</h2.version>
+        <spring-version>5.2.6.RELEASE</spring-version>
```
Comment (vinnybod) [inline pom.xml:58]: I think the spring version is handled by spring boot. If you mix these you could end up with runtime conflicts

Comment (brianksorensen) [reply pom.xml:58]: I pulled the branch and checked.  5.2.6.RELEASE is the version of the spring-jdbc, which references versions 5.1.3.RELEASE of spring-core, spring-bean, and spring-tx.  So label wise, tag should probably be spring-jdbc-version .

Comment (maddapper) [reply pom.xml:58]: Right, good point there.  I'll change spring-version to spring-jdbc-version.

Comment (maddapper) [reply pom.xml]: Thanks for the review gents.

**Pull request merged**